src/dbnode/storage/bootstrap/types.go

```diff
@@ -23,14 +23,20 @@ package bootstrap
 import (
 	"time"
 
-	"github.com/m3db/m3/src/dbnode/client"
 	"github.com/m3db/m3/src/dbnode/persist"
 	"github.com/m3db/m3/src/dbnode/storage/bootstrap/result"
 	"github.com/m3db/m3/src/dbnode/storage/namespace"
 	"github.com/m3db/m3/src/dbnode/topology"
 	xtime "github.com/m3db/m3x/time"
 )
 
+// TopologyMapProvider is an interface that can provide
+// a topology map.
+type TopologyMapProvider interface {
```

Comment:
nit: move to `topology` package. Can call it `topology.MapProvider` or `topology.MapResolver`

Reply:
Seemed nice to have it close to where it is used, but don't have strong feelings about it

Reply:
i just don't like the import dependency leaving this here creates. `storage/cluster` and and `storage/bootstrap` both need this type. Better place for it is `topology` so both can depend on it, rather than have `cluster` depend on `bootstrap` as it would be if you left here

Reply:
fair, fixed